packages/schematics/src/tslint/nxEnforceModuleBoundariesRule.ts

```diff
@@ -55,7 +55,13 @@ class EnforceModuleBoundariesWalker extends Lint.RuleWalker {
     private roots: string[]
   ) {
     super(sourceFile, options);
-    this.roots = [...roots].sort((a, b) => a.length - b.length);
+    //
+    // TODO
+    // remove sort completely in favor of appending '/' at libraryRoot()?
+    // sort logic does not work for lazyLoaded, so one logic to check for name collision?
+    //
+    // fix: sort from longest to shortest
+    this.roots = [...roots].sort((a, b) => b.length - a.length);
```

Comment:
could you add a test for this change?

Reply:
I can add a test case where I have the libs `mylib` and `mylibtwo`. I "create" a file at `/dev/workspace/libs/mylibtwo/mycomp.ts` and do a relative import like `./mycomptwo`. this should pass and not error with `library imports must start with @myworkspace/`. did you mean something like that?

the test `should not error when relatively importing the same library` did pass prior to this pr because the file was created in the shorter named lib and the shorter named lib came first because of the wrong sort ...

besides all of that, I did some refactoring concerning the lint rule (and added glob support for the `allow` option). do you mind having a look [here](https://github.com/skydever/nx/pull/7)? Is it good/bad? The concept was to evaluate the `root` of both source and target at the beginning, then the checks seemed easier...